### ISSUE TITLE
Confirm user's email address after registration

### ISSUE THREAD

**Issue by bajiat** [apinf/api-umbrella-dashboard#233]
When user sign-ups to Apinf, the user gets a confirmation email. There should be a link in the email for the purpose of confirming that it is a valid and working email address.
Note: user story missing.

# Task
## Research
* [x] Determine how other Meteor projects send user emails
* [x] Search for any verification email plugins, tutorials, or examples
* [x] Choose how to implement the verification email workflow

# Implement
* [x] Install any necessary packages for verifying user emails
* [x] Create verification email template(s)
* [x] Update registration flow to include email verification step

# Resources
This may already be handled by the useraccounts package [basic configuration object](https://github.com/meteor-useraccounts/core/blob/master/Guide.md#options), which are installed in our project:

```javascript
AccountsTemplates.configure({
  ...
    // Behavior
    sendVerificationEmail: false,
  ...
```

**Comment by mauriciovieira** [comment]
@bajiat I can take this one.
I think it's related to  #343

**Comment by bajiat** [comment]
Ok!

**Comment by brylie** [comment]
Co-pilot: @brylie

**Comment by jykae** [comment]
Found right places in the code. I tried switching sendVerificationEmail to true and commented out mail sending call on Account creation. Got email, looks like this: https://goo.gl/F8Qu2n

It still uses mailgun for smtp, as I didn't change that from settings. Template for verification email needs to be created. Verification link works.

Many Meteor projects seem to use Mailgun for email. That is fine for development, but in the production deployment application server itself should be able to send email without 3rd party service.

Possible alternative is to have Postfix installed on deployment server, http://www.hiddentao.com/archives/2014/12/18/meteor-localhost-smtp-using-postfix-on-ubuntu/ This however expects Ubuntu to be platform for deployment.
